tsfile/src/test/java/org/apache/iotdb/tsfile/write/MetadataIndexConstructorTest.java

```diff
@@ -237,6 +235,19 @@ private void test(String[] devices, int[][] vectorMeasurement, String[][] single
             correctFirstMeasurements.get(j).get(i * conf.getMaxDegreeOfIndexNode()));
       }
     }
+    try (TsFileSequenceReader reader = new TsFileSequenceReader(FILE_PATH)) {
+      Iterator<List<Path>> iterator = reader.getPathsIterator();
+      int idx = 0;
+      while (iterator.hasNext()) {
+        for (Path actualPath : iterator.next()) {
+          assertEquals(actualPath.getFullPath(), correctPaths.get(idx));
+          idx++;
+        }
+      }
```

Comment:
I think we may need to assert the `idx` value finally, as the paths size returned by the iterator may be less than the size of `correctPaths` and the test can't find this bug if it really happens.
Add `Assert.assertEquals(correctPaths.size(), idx);`

Reply:
Great point! Thanks a lot.